### PR TITLE
Add missing methods methods in aiofiles

### LINKED ISSUE #4714: Invalid aiofiles stubs
Even after PR #4650, the stubs for aiofiles are incorrect.

The stubs still don't include wrapped methods for AsyncBufferedReader, AsyncBufferedIOBase, AsyncBufferedReader and AsyncFileIO. Something like this doesn't type check, since, according to current stubs, AsyncBufferedReader has no read method:

```python
import aiofiles

async def print_file(path: str) -> None:
    async with aiofiles.open(path, mode="r") as f:
        print(await f.read())
```

And also, there's no `aiofiles._os` module, it should be `aiofiles.os`.

### PR DESCRIPTION
**Pull request by MKuranowski**
This PR adds multiple missing methods in the `aiofiles` package.

Objects returned by `aiofiles.open` have asynchronous methods corresponding to synchronous methods on usual file-like objects.

`aiofiles` "defines" those methods with a decorator directly on AsyncBufferedIOBase, AsyncBufferedReader, AsyncFileIO and AsyncTextIOWrapper classes, so I just copied function signatures from the built-in `io` module stubs to those classes. However, it's not always possible to determine the `buffering` attribute, so I had to create a fake `_UnknownAsyncBinaryIO` class, which is an intersection of buffered and unbuffered file-like objects.

I also made said async classes generic to create a proper signature for the `name` property.

The `aiofiles._os` doesn't exist, instead `aiofiles.os` is available - I renamed the corresponding stub file.

Fixes #4714.

### CODE REVIEW

File: third_party/3/aiofiles/threadpool/binary.pyi [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+from typing import Generic, Iterable, List, Optional, TypeVar, Union
+
 from ..base import AsyncBase
 
-class AsyncBufferedIOBase(AsyncBase[bytes]): ...
-class AsyncBufferedReader(AsyncBufferedIOBase): ...
-class AsyncFileIO(AsyncBase[bytes]): ...
+_FileName = TypeVar("_FileName", bound=Union[AnyPath, int])
+
+class _UnknownAsyncBinaryIO(AsyncBase[bytes], Generic[_FileName]):
```
Comment (JelleZijlstra) [inline third_party/3/aiofiles/threadpool/binary.pyi:9]: I'm not sure making this generic over _Filename is worth it; it's slightly more precise for the `.filename` property, but now forces everyone who uses one of these classes as a type annotation to add a type argument.

File: third_party/3/aiofiles/threadpool/__init__.pyi [hunk 2]
```diff
@@ -33,37 +38,37 @@ def open(
...
     file: _OpenFile,
-    mode: OpenBinaryModeWriting,
+    mode: Union[OpenBinaryModeReading, OpenBinaryModeUpdating],
     buffering: Literal[-1, 1] = ...,
     encoding: None = ...,
     errors: None = ...,
     newline: None = ...,
     closefd: bool = ...,
     opener: Optional[_Opener] = ...,
-    *,
```
[the remark this answers is not in the document]
Comment (MKuranowski) [reply third_party/3/aiofiles/threadpool/__init__.pyi:49]: Oh, that's a mistake. Pushed a fix.

File: third_party/3/aiofiles/threadpool/binary.pyi (hunk 1 above)
Comment (MKuranowski) [reply third_party/3/aiofiles/threadpool/binary.pyi:9]: > now forces everyone who uses one of these classes as a type annotation to add a type argument.

Is it true? I use pyright and it doesn't complain about not setting a generic type.

```python
async def(file: AsyncTextIOWrapper):
    name = file.name
    # name is auto-detected to be of type: str | bytes | _PathLike[str] | _PathLike[bytes] | int
```

Of course, if other type checkers complain, I can annotate `name` property as `-> Union[AnyPath, int]`.

Comment (JelleZijlstra) [reply third_party/3/aiofiles/threadpool/binary.pyi:9]: Mypy does this if you use the `--disallow-any-generics` option.

Comment (MKuranowski) [reply third_party/3/aiofiles/threadpool/binary.pyi]: Ok, removed that.

**Pull request merged**